src/amp-story-player/amp-story-player-impl.js

```diff
@@ -611,11 +611,34 @@ export class AmpStoryPlayer {
       return;
     }
 
-    this.currentIdx_ = storyIdx;
+    const adjacentStoriesIdx = this.iframePool_.findAdjacent(
```

Comment:
This looks like the only use of `findAdjacent`.  Given that we hope for this to be a common path, should we just generate the array in this order in the first place, to avoid multiple operations to the same array?  The first operation (in sorted order) will never be used as is.

Reply:
We use this order to make sure that the `storyIdsWithIframe_` inside the iframe pool stays sorted when calling `rotateFirst/Last`, so that we always evict the first iframe and assign it to the next upcoming story when navigating forwards. But since `rotateFirst/Last` is not a frequent operation I think it's fine to move it there. I made this change in the latest commit, PTAL.